### PR TITLE
Sample for ConvertType transform estimator

### PR DESCRIPTION
**Pull request by abgoswam**
Towards #1209

- fix bug in ConvertType   (exception when trying to convert from Bool => Int)
- add sample for ConvertType

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public static class ConvertTypeTransform
+    {
+        private sealed class InputData
+        {
+            public bool Survived;
+        }
+
+        private sealed class TransformedData
+        {
+            public bool Survived { get; set; }
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: >{ get; set; } [](start = 33, length = 13)

do those need to be properties?

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            // Let's train our pipeline, and then apply it to the same data.
+            var transformer = pipeline.Fit(data);
+            var transformedData = transformer.Transform(data);
+
+            // Display original column 'Survived' (boolean) and converted column 'SurvivedInt32' (Int32)
+            var convertedData = mlContext.Data.CreateEnumerable<TransformedData>(transformedData, true);
+            foreach (var item in convertedData)
+            {
+                Console.WriteLine("A:{0}  Aconv:{1}", item.Survived, item.SurvivedInt32);
+            }
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: Put in comments the output of this WriteLine.

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        private sealed class TransformedData
+        {
+            public bool Survived { get; set; }
+
+            public Int32 SurvivedInt32 { get; set; }
+        }
+
+        public static void Example()
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: >Example [](start = 27, length = 7)

the sample needs to be linked from the actual source it illustrates. 

Checkout the <example> node on the other extension methods.

**Comment by shmoradims** [comment]
> using System;

Please move this file to a path similar to it's access path from MLContext: Dynamic/Transforms/Conversion/ConvertType.cs

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs:1 in 5793031. [](commit_id = 5793031aa0951f262418522701dbf388d158d785, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class ConvertTypeTransform
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: >ConvertTypeTransform [](start = 24, length = 20)

Please rename this to ConvertType (identical to the API extension method that it correspond to)

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs (hunk 1 above)
Comment (abgoswam) [reply docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: setting it to properties helps me get away with warning about uninitialized field

---
In reply to: [261046835](https://github.com/dotnet/machinelearning/pull/2781#discussion_r261046835) [](ancestors = 261046835)

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs (hunk 2 above)
Comment (abgoswam) [reply docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: fixed

---
In reply to: [261046921](https://github.com/dotnet/machinelearning/pull/2781#discussion_r261046921) [](ancestors = 261046921)

File: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs (hunk 3 above)
Comment (abgoswam) [reply docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs]: fixed

---
In reply to: [261047048](https://github.com/dotnet/machinelearning/pull/2781#discussion_r261047048) [](ancestors = 261047048)

**Comment by abgoswam** [comment]
> using System;

fixed

---
In reply to: [468489940](https://github.com/dotnet/machinelearning/pull/2781#issuecomment-468489940) [](ancestors = 468489940)

---
Refers to: docs/samples/Microsoft.ML.Samples/Dynamic/ConvertType.cs:1 in 5793031. [](commit_id = 5793031aa0951f262418522701dbf388d158d785, deletion_comment = False)

File: test/Microsoft.ML.Tests/Transformers/ConvertTests.cs [hunk 5]
```diff
@@ -98,7 +98,7 @@ public void TestConvertWorkout()
                   new TestPrimitiveClass()
                 {
                     AA = new []{"0", "1"},
-                    AB = new []{false, true},
+                    AB = new []{true, false},
```
Comment (abgoswam) [inline test/Microsoft.ML.Tests/Transformers/ConvertTests.cs]: >true, false} [](start = 32, length = 12)

p.s. I modified the order just to ensure that the 2nd row is different from the 1st row above.

**Pull request merged**